app/jobs/push_opportunity_to_salesforce_job.rb

```diff
@@ -47,7 +47,16 @@ def perform(opp)
     rescue => e
       Sentry.capture_exception(e)
     end
-
     opportunity
   end
+
+  def calculate_start_date
+    year = Date.today.year
+    between = Date.today.between?(Date.strptime(year.to_s + '-04-01', '%Y-%m-%d'), Date.strptime(year.to_s + '-11-01', '%Y-%m-%d'))
+    if between
+      Date.strptime(year.to_s + '-08-15', '%Y-%m-%d')
+    else
+      Date.strptime(year.to_s + '-01-06', '%Y-%m-%d')
```

Comment:
Just a suggestion - might make this a little more readable if you use the ternary operator:

```
def calculate_start_date
  year = Date.today.year
  between_april_and_november = Date.today.between?(Date.strptime(year.to_s + '-04-01', '%Y-%m-%d'), Date.strptime(year.to_s + '-11-01', '%Y-%m-%d'))

  fall_semester = Date.strptime(year.to_s + '-08-15', '%Y-%m-%d')
  spring_semester = Date.strptime(year.to_s + '-01-06', '%Y-%m-%d')

  between_april_and_november ? fall_semester : spring_semester
end

```

Reply:
Done